lib/widgets/currency_converter_dialog.dart

```diff
@@ -0,0 +1,255 @@
+                      controller: _fiatAmountController,
+                      validator: (_) {
+                        if (widget.validatorFn != null) {
+                          return widget.validatorFn(_satoshies);
+                        }
+                        return null;
+                      },
+                      style: theme.alertStyle),
+                ),
+                Padding(
+                  padding: const EdgeInsets.only(top: 24.0),
+                  child: Column(
+                    children: <Widget>[
+                      Text("${_fiatAmountController.text.isNotEmpty ? _currency.format(_satoshies, includeSymbol: false) : 0} ${_currency.symbol}",
+                          style: theme.headline.copyWith(fontSize: 16.0)),
+                      _buildExchangeRateLabel()
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            actions: _buildActions(),
+          );
+        });
+  }
+
+  List<Widget> _buildActions() {
+    List<Widget> actions = [new FlatButton(onPressed: () => Navigator.pop(context), child: new Text("Cancel", style: theme.buttonStyle))];
+
+    // Show done button only when the converted amount is bigger than 0
+    if (_fiatAmountController.text.isNotEmpty && _satoshies > 0) {
+      actions.add(new FlatButton(
+          onPressed: () {
+            if (_formKey.currentState.validate()) {
+              widget._onConvert(_currency.format(_satoshies, includeSymbol: false, userInput: true));
+              Navigator.pop(context);
+            }
+          },
+          child: new Text("Done", style: theme.buttonStyle)));
+    }
+    return actions;
+  }
+
+  _updateExchangeLabel(double exchangeRate) {
+    if (_exchangeRate != exchangeRate) {
+      // Blink exchange rate label when exchange rate changes (also switches between fiat currencies)
+      if (_exchangeRate != null && !_controller.isAnimating) {
+        _controller.forward();
+      }
+      _exchangeRate = exchangeRate;
+    }
+  }
+
+  _convertCurrency() {
+    var _bitcoinEquivalent = _fiatAmountController.text.isNotEmpty ? _fiatCurrency.convertToBTC(double.parse(_fiatAmountController.text ?? 0)) : 0;
+    setState(() {
+      _satoshies = Int64((_bitcoinEquivalent * 100000000).toInt());
+    });
+  }
+
+  Widget _buildExchangeRateLabel() {
+    // Empty string widget is returned so that the dialogs height is not changed when the exchange rate is shown
+    return _exchangeRate == null
+        ? Text("", style: theme.smallTextStyle)
+        : Text("1 BTC = $_exchangeRate ${_fiatCurrency.currencyData.shortName}",
+            style: theme.smallTextStyle.copyWith(color: _colorAnimation.value));
+  }
+
+  _selectFiatCurrency(shortName) {
+    setState(() {
+      int _oldFractionSize = _fiatCurrency.currencyData.fractionSize;
+      _fiatCurrency = _fiatConversionList.firstWhere((f) => f.currencyData.shortName == shortName);
```

Comment:
Also here no need to count on this state variable as it is not needed. Instead I suggest to pass the `AccountModel` to this method that is called from the `build` method.